File: app/src/main/java/com/nacoda/moviesmvvm/util/Injection.java
package com.nacoda.moviesmvvm.util;

import android.content.Context;
import com.nacoda.moviesmvvm.data.source.MoviesDataSource;
import com.nacoda.moviesmvvm.data.source.MoviesRepository;
import com.nacoda.moviesmvvm.data.source.local.MoviesLocalDataSource;
import com.nacoda.moviesmvvm.data.source.remote.MoviesRemoteDataSource;

/**
 * Created by Mayburger on 1/9/18.
 */

class Injection {

    static volatile Injection INSTANCE;

    final MoviesRepository provideMoviesRepository(Context mContext) {
        MoviesRepository.Companion moviesRepository = MoviesRepository.Companion;
        MoviesDataSource moviesDataSource = new MoviesRemoteDataSource();
        return moviesRepository.getInstance(moviesDataSource, new MoviesLocalDataSource());
    }

    static {
        Injection var0 = new Injection();
        INSTANCE = var0;
    }
}

File: app/src/main/java/com/nacoda/moviesmvvm/util/Injection.kt
package com.nacoda.moviesmvvm.util

import android.content.Context
import android.preference.PreferenceManager
import com.nacoda.moviesmvvm.data.source.MoviesRepository
import com.nacoda.moviesmvvm.data.source.local.MoviesLocalDataSource
import com.nacoda.moviesmvvm.data.source.remote.MoviesRemoteDataSource

/**
 * Created by irfanirawansukirman on 04/12/17.
 */
object Injection {

    fun provideMoviesRepository(mContext: Context) =
            MoviesRepository.getInstance(MoviesRemoteDataSource(),
                    MoviesLocalDataSource.getInstance(PreferenceManager.getDefaultSharedPreferences(mContext)))
}